src/app/rosetta/lib/account.ml

```diff
@@ -113,158 +117,153 @@ module Sql = struct
-        let initial_minimum_balance =
-          MinaCurrency.Balance.of_string timing_info.initial_minimum_balance
-        in
-        Mina_base.Account.incremental_balance_between_slots ~start_slot ~end_slot
-          ~cliff_time ~cliff_amount ~vesting_period ~vesting_increment
-          ~initial_minimum_balance
-      in
-      let%bind (liquid_balance, nonce) =
-        match (last_relevant_command_info_opt, timing_info_opt) with
-        | None, None ->
-          (* We've never heard of this account, at least as of the
-             block_identifier provided This means they requested a
-             block from before account creation; this is ambiguous in
-             the spec but Coinbase confirmed we can return 0.
-             https://community.rosetta-api.org/t/historical-balance-requests-with-block-identifiers-from-before-account-was-created/369 *)
-          Deferred.Result.return (0L, UInt64.zero)
-        | Some (_, _, last_relevant_command_balance, nonce), None ->
-          (* This account has no special vesting, so just use its last
-             known balance from the command.*)
-          Deferred.Result.return (last_relevant_command_balance, UInt64.of_int64 nonce)
-        | None, Some timing_info ->
-          (* This account hasn't seen any transactions but was in the
-             genesis ledger, so use the balance obtained from GraphQL. *)
-           let balance_at_genesis : int64 =
-             Get_balance.(Option.value
-               ~default:account.balance.total
-               account.balance.liquid)
-           |> Unsigned.UInt64.to_int64
-          in
-          let incremental_balance_since_genesis : UInt64.t =
-            compute_incremental_balance timing_info
-              ~start_slot:(UInt32.of_int 0)
-          in
-          Deferred.Result.return
-            ( UInt64.Infix.(
-                  UInt64.of_int64 balance_at_genesis
-                  + incremental_balance_since_genesis)
-              |> UInt64.to_int64, UInt64.zero)
-        | ( Some
-              (_, last_relevant_command_global_slot_since_genesis
-              , last_relevant_command_balance, nonce )
-          , Some timing_info ) ->
-          (* This block was in the genesis ledger and has been
-             involved in at least one user or internal command. We need *
-             to compute the change in its balance between the most recent
-             command and the start block (if it has vesting * it may have
-             changed). *)
-          let incremental_balance_between_slots =
-            compute_incremental_balance timing_info
-              ~start_slot:
+       let%bind timing_info_opt =
+         Archive_lib.Processor.Timing_info.find_by_account_identifier_id_opt
+           (module Conn)
+           account_identifier_id
+         |> Errors.Lift.sql ~context:"Finding timing info"
+       in
+       let end_slot =
+         UInt32.of_int
+           (Int.of_int64_exn requested_block_global_slot_since_genesis)
```

Comment:
This is existing code but I think the following should be better. It is simpler and can avoid unneeded exceptions on 32 bits systems.

```suggestion
         UInt32.of_int64 requested_block_global_slot_since_genesis
```